### PR TITLE
feat: add repeatWithViewLifecycle and launchAndRepeatWithViewLifecycle 

### PR DESCRIPTION
**Pull request by atiqaraj**

### CODE REVIEW

File: app/src/main/java/com/monstarlab/features/login/LoginFragment.kt [hunk 1]
```diff
@@ -36,15 +38,21 @@ class LoginFragment : Fragment(R.layout.fragment_login) {
...
+            }
+
+            launch {
+                viewModel.loadingFlow.collect { loading ->
+                    TransitionManager.beginDelayedTransition(binding.root)
+                    binding.loginEmailEditText.isEnabled = !loading
+                    binding.loginPasswordEditText.isEnabled = !loading
+                    binding.loginButton.isVisible = !loading
+                }
+            }
```
Comment (johsoe) [inline app/src/main/java/com/monstarlab/features/login/LoginFragment.kt:55]: Naming aside, I dont see how this is nicer in any way?

Comment (atiqmlbd) [reply app/src/main/java/com/monstarlab/features/login/LoginFragment.kt:55]: yeah, this is probably not nicer. but based on the documentation of `flowWithLifecycle`
https://androidx.tech/artifacts/lifecycle/lifecycle-runtime-ktx/2.4.0-alpha01-source/androidx/lifecycle/FlowExt.kt.html
it's better this way
`Tip: If multiple flows need to be collected using flowWithLifecycle, consider using the Lifecycle.repeatOnLifecycle API to collect from all of them using a different launch per flow instead. That's more efficient and consumes less resources as no hot flows are created.`

File: app/src/main/java/com/monstarlab/features/resources/ResourceFragment.kt [hunk 2]
```diff
@@ -19,28 +21,36 @@ class ResourceFragment : Fragment(R.layout.fragment_resource) {
...
+            {
+                viewModel.loadingFlow.collect { loading ->
+                    TransitionManager.beginDelayedTransition(binding.root)
+                    binding.resourceRecyclerView.visibility =
+                        if (loading) View.GONE else View.VISIBLE
+                    binding.resourceProgressBar.visibility =
+                        if (loading) View.VISIBLE else View.GONE
+                }
+            }
+        )
```
[the remark this answers is not in the document]
Comment (atiqmlbd) [reply app/src/main/java/com/monstarlab/features/resources/ResourceFragment.kt:53]: the same, here It's just another variant hiding launch every time

**Comment by levinzonr** [comment]
Closed
Reason: Since we are using Jetpack Compose we are not gonna need this

**Pull request closed without merging**